diff -Naur mpdecimal-4.0.0-orig/config.sub mpdecimal-4.0.0/config.sub
--- mpdecimal-4.0.0-orig/config.sub	2024-01-10 00:25:00
+++ mpdecimal-4.0.0/config.sub	2025-03-09 19:38:32
@@ -1127,7 +1127,7 @@
 	xscale-* | xscalee[bl]-*)
 		cpu=`echo "$cpu" | sed 's/^xscale/arm/'`
 		;;
-	arm64-* | aarch64le-*)
+	arm64-* | arm64_32-* | aarch64le-*)
 		cpu=aarch64
 		;;
 
@@ -1736,7 +1736,7 @@
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
 	     | midnightbsd* | amdhsa* | unleashed* | emscripten* | wasi* \
 	     | nsk* | powerunix* | genode* | zvmoe* | qnx* | emx* | zephyr* \
-	     | fiwix* | mlibc* | cos* | mbr* )
+	     | fiwix* | mlibc* | cos* | mbr* | macabi )
 		;;
 	# This one is extra strict with allowed versions
 	sco3.2v2 | sco3.2v[4-9]* | sco5v6*)
@@ -1793,6 +1793,10 @@
 		;;
 	*-eabi* | *-gnueabi*)
 		;;
+	ios*-simulator | tvos*-simulator | watchos*-simulator)
+		;;
+	ios*-macabi)
+	;;
 	none-coff* | none-elf*)
 		# None (no kernel, i.e. freestanding / bare metal),
 		# can be paired with an output format "OS"
