diff -ur xz-5.6.4-orig/build-aux/config.sub xz-5.6.4/build-aux/config.sub
--- xz-5.6.4-orig/build-aux/config.sub	2025-01-23 17:58:05
+++ xz-5.6.4/build-aux/config.sub	2025-04-18 11:48:46
@@ -1194,7 +1194,7 @@
 	xscale-* | xscalee[bl]-*)
 		cpu=`echo "$cpu" | sed 's/^xscale/arm/'`
 		;;
-	arm64-* | aarch64le-*)
+	arm64-* | aarch64le-* | arm64_32-*)
 		cpu=aarch64
 		;;
 
@@ -2033,6 +2033,7 @@
 	| lynxos* \
 	| mach* \
 	| macos* \
+	| macabi \
 	| magic* \
 	| mbr* \
 	| midipix* \
@@ -2148,6 +2149,7 @@
 	| winnt* \
 	| xenix* \
 	| xray* \
+	| xros* \
 	| zephyr* \
 	| zvmoe* )
 		;;
@@ -2248,6 +2250,8 @@
 	rtmk-nova-)
 		;;
 	*-eabi*- | *-gnueabi*-)
+		;;
+	ios*-simulator- | ios*-macabi- | tvos*-simulator- | watchos*-simulator- | xros*-simulator-)
 		;;
 	none--*)
 		# None (no kernel, i.e. freestanding / bare metal),
